src/views/Predictions/components/RoundCard/SetPositionCard.tsx

```diff
@@ -86,30 +72,46 @@ const SetPositionCard: React.FC<SetPositionCardProps> = ({ position, togglePosit
   const { toastError } = useToast()
   const predictionsContract = usePredictionsContract()
 
-  const balanceDisplay = getBnbAmount(bnbBalance).toString()
-  const maxBalance = getBnbAmount(bnbBalance.gt(dust) ? bnbBalance.minus(dust) : bnbBalance).toNumber()
+  const balanceDisplay = useMemo(() => {
+    return getBnbAmount(bnbBalance).toString()
+  }, [bnbBalance])
+  const maxBalance = useMemo(() => {
+    return getBnbAmount(bnbBalance.gt(dust) ? bnbBalance.minus(dust) : bnbBalance)
+  }, [bnbBalance])
+  const minBetAmountBalance = useMemo(() => {
+    return getBnbAmount(minBetAmount)
+  }, [minBetAmount])
+
   const valueAsBn = new BigNumber(value)
 
-  const percentageOfMaxBalance = valueAsBn.div(maxBalance).times(100).toNumber()
-  const percentageDisplay = getPercentDisplay(percentageOfMaxBalance)
   const showFieldWarning = account && valueAsBn.gt(0) && errorMessage !== null
-  const minBetAmountBalance = getBnbAmount(minBetAmount).toNumber()
 
-  const handleChange = (input) => {
-    setValue(input)
-  }
+  const [percent, setPercent] = useState(0)
```

Comment:
Why keep percent in state? It is easily derived from from the current value and the user's balance.

Reply:
see my comment https://github.com/pancakeswap/pancake-frontend/pull/1454#issuecomment-857971554